src/partition/repart.c

```diff
@@ -4131,15 +4132,15 @@ static int parse_argv(int argc, char *argv[]) {
                 { "version",           no_argument,       NULL, ARG_VERSION           },
                 { "no-pager",          no_argument,       NULL, ARG_NO_PAGER          },
                 { "no-legend",         no_argument,       NULL, ARG_NO_LEGEND         },
-                { "dry-run",           required_argument, NULL, ARG_DRY_RUN           },
+                { "dry-run",           optional_argument, NULL, ARG_DRY_RUN           },
                 { "empty",             required_argument, NULL, ARG_EMPTY             },
                 { "discard",           required_argument, NULL, ARG_DISCARD           },
-                { "factory-reset",     required_argument, NULL, ARG_FACTORY_RESET     },
+                { "factory-reset",     optional_argument, NULL, ARG_FACTORY_RESET     },
                 { "can-factory-reset", no_argument,       NULL, ARG_CAN_FACTORY_RESET },
                 { "root",              required_argument, NULL, ARG_ROOT              },
                 { "image",             required_argument, NULL, ARG_IMAGE             },
                 { "seed",              required_argument, NULL, ARG_SEED              },
-                { "pretty",            required_argument, NULL, ARG_PRETTY            },
+                { "pretty",            optional_argument, NULL, ARG_PRETTY            },
```

Comment:
hmpf. the `optional_argument` stuff in getop_long() is pretty crappy.

The thing is that for `no_argument` and `required_argument` it's very clear what this is:

```
foo --bar waldo
```

(i.e. for `no_argument` there's one switch `--bar` and one positional argument `waldo`. And for `required_argument` it's one switch with an argument `--bar=waldo`, with no positional argument.)

But for `optional_argument` it's weird. Can you answer correctly how getop_long() will interpret the line above without looking it up? 

(For short options it's even more confusing btw)

Because behaviour is so weird we actually ended up overriding getopt_long()'s own handling in some cases, for example journalctl's `-n` switch, where we manually modify optind in order to do something people are less confused about...

Because of that we kinda stopped using `optional_argument` switch in new code.

I am not sure we should start here again...

Reply:
I figured there'd be a reason behind this. Your explanation makes sense, so let's close this. It'd be nice if we could make this work but that seems like it would require more involved changes to argument parsing